tests/test_framework.py

```diff
@@ -164,12 +228,134 @@ def test_panoptes_resource_set(self):
+            resource_key = "plugin|test|site|test|class|test|subclass|test|type|test|id|test|endpoint|test"
+            resource_value = panoptes_resource_store.get_resource(resource_key)
+            self.assertEqual(self.__panoptes_resource, resource_value)
+
+            panoptes_resource_2 = PanoptesResource(resource_site='test', resource_class='test',
+                                                   resource_subclass='test',
+                                                   resource_type='test', resource_id='test2', resource_endpoint='test',
+                                                   resource_plugin='test',
+                                                   resource_ttl=RESOURCE_MANAGER_RESOURCE_EXPIRE,
+                                                   resource_creation_timestamp=_TIMESTAMP)
+            panoptes_resource_store.add_resource("test_plugin_signature", panoptes_resource_2)
+            self.assertIn(self.__panoptes_resource, panoptes_resource_store.get_resources())
+            self.assertIn(panoptes_resource_2, panoptes_resource_store.get_resources())
+
+            panoptes_resource_store.delete_resource("test_plugin_signature", panoptes_resource_2)
+            self.assertNotIn(panoptes_resource_2, panoptes_resource_store.get_resources())
+
+            panoptes_resource_3 = PanoptesResource(resource_site='test', resource_class='test',
+                                                   resource_subclass='test',
+                                                   resource_type='test', resource_id='test3', resource_endpoint='test',
+                                                   resource_plugin='test3',
+                                                   resource_ttl=RESOURCE_MANAGER_RESOURCE_EXPIRE,
+                                                   resource_creation_timestamp=_TIMESTAMP)
+            panoptes_resource_store.add_resource("test_plugin_signature", panoptes_resource_3)
+            self.assertIn(panoptes_resource_3, panoptes_resource_store.get_resources(site='test', plugin_name='test3'))
+            self.assertNotIn(self.__panoptes_resource,
+                             panoptes_resource_store.get_resources(site='test', plugin_name='test3'))
+
+            # Test key not found
+            mock_find_keys = Mock(
+                return_value=['dummy',
+                              'plugin|test|site|test|class|test|subclass|test|type|test|id|test|endpoint|test'])
+            with patch('yahoo_panoptes.framework.resources.PanoptesKeyValueStore.find_keys',
+                       mock_find_keys):
+                self.assertEqual(1, len(panoptes_resource_store.get_resources()))
+
+            # Test resource store methods raise correct errors
+            mock_get = Mock(side_effect=Exception)
+            with patch('yahoo_panoptes.framework.resources.PanoptesKeyValueStore.get', mock_get):
+                with self.assertRaises(Exception):
```

Comment:
Could you please change get_resource to raise a PanoptesResourceError instead of a bare Exception in get_resource